Libraries/Components/DrawerAndroid/AndroidDrawerLayoutNativeComponent.js

```diff
@@ -0,0 +1,129 @@
+     *   - 'none' (the default), drags do not dismiss the keyboard.
+     *   - 'on-drag', the keyboard is dismissed when a drag begins.
+     */
+    keyboardDismissMode?: ?('none' | 'on-drag'),
+  
+    /**
+     * Specifies the background color of the drawer. The default value is white.
+     * If you want to set the opacity of the drawer, use rgba. Example:
+     *
+     * ```
+     * return (
+     *   <DrawerLayoutAndroid drawerBackgroundColor="rgba(0,0,0,0.5)">
+     *   </DrawerLayoutAndroid>
+     * );
+     * ```
+     */
+    drawerBackgroundColor: ColorValue,
+  
+    /**
+     * Specifies the side of the screen from which the drawer will slide in.
+     */
+    drawerPosition: ?number,
+  
+    /**
+     * Specifies the width of the drawer, more precisely the width of the view that be pulled in
+     * from the edge of the window.
+     */
+    drawerWidth?: ?number,
+  
+    /**
+     * Specifies the lock mode of the drawer. The drawer can be locked in 3 states:
+     * - unlocked (default), meaning that the drawer will respond (open/close) to touch gestures.
+     * - locked-closed, meaning that the drawer will stay closed and not respond to gestures.
+     * - locked-open, meaning that the drawer will stay opened and not respond to gestures.
+     * The drawer may still be opened and closed programmatically (`openDrawer`/`closeDrawer`).
+     */
+    drawerLockMode?: ?('unlocked' | 'locked-closed' | 'locked-open'),
+  
+    /**
+     * Function called whenever there is an interaction with the navigation view.
+     */
+    onDrawerSlide?: ?(event: DrawerSlideEvent) => mixed,
+  
+    /**
+     * Function called when the drawer state has changed. The drawer can be in 3 states:
+     * - Idle, meaning there is no interaction with the navigation view happening at the time
+     * - Dragging, meaning there is currently an interaction with the navigation view
+     * - Settling, meaning that there was an interaction with the navigation view, and the
+     * navigation view is now finishing its closing or opening animation
+     */
+    onDrawerStateChanged?: ?(state: DrawerStates) => mixed,
+  
+    /**
+     * Function called whenever the navigation view has been opened.
+     */
+    onDrawerOpen?: ?() => mixed,
+  
+    /**
+     * Function called whenever the navigation view has been closed.
+     */
+    onDrawerClose?: ?() => mixed,
+  
+    /**
+     * The navigation view that will be rendered to the side of the screen and can be pulled in.
+     */
+    renderNavigationView: () => React.Element<any>,
+  
+    /**
+     * Make the drawer take the entire screen and draw the background of the
+     * status bar to allow it to open over the status bar. It will only have an
+     * effect on API 21+.
+     */
+    statusBarBackgroundColor?: ?ColorValue,
+  
+    children?: React.Node,
+    style?: ?ViewStyleProp,
+  |}>;
+
+type NativeProps = $ReadOnly<{|
+    ...$Diff<
```

Comment:
I don’t think we should use Diff here, can we just change the props definition? Also, feel free to just merge the two objects, they are all Native props! :-)

Reply:
Do you mean merging Props type with NativeProps into something like this?


``` es6
/**
* Before
*/
type NativeProps = $ReadOnly<{|
    ...$Diff<
      Props,
      $ReadOnly<{onDrawerStateChanged?: ?(state: DrawerStates) => mixed}>,
    >,
    onDrawerStateChanged?: ?(state: DrawerStateEvent) => mixed,
  |}>;

/**
* After
*/
type NativeProps = $ReadOnly<{|
    onDrawerStateChanged?: ?((state: DrawerStateEvent) => mixed), 
    children?: Node,
    drawerBackgroundColor: ColorValue,
    drawerLockMode?: ?("unlocked" | "locked-closed" | "locked-open"),
    drawerPosition: ?number,
    drawerWidth?: ?number,
    keyboardDismissMode?: ?("none" | "on-drag"),
    onDrawerClose?: ?(() => mixed),
    onDrawerOpen?: ?(() => mixed),
    onDrawerSlide?: ?((event: DrawerSlideEvent) => mixed),
    renderNavigationView: () => Element<any>,
    statusBarBackgroundColor?: ?ColorValue,
    style?: ?ViewStyleProp
  |}>
```


I'm also not sure about `onDrawerStateChanged`, should the type be `?((state: DrawerStateEvent) => mixed)`? 


I tried reading about how the `...$Diff` utility type works, but I'm still confused.

 If I got it wrong, any help would be welcome at this point 🙏

Reply:
Do you mean merging Props type with NativeProps into something like this?


``` es6
/**
* Before
*/
type NativeProps = $ReadOnly<{|
    ...$Diff<
      Props,
      $ReadOnly<{onDrawerStateChanged?: ?(state: DrawerStates) => mixed}>,
    >,
    onDrawerStateChanged?: ?(state: DrawerStateEvent) => mixed,
  |}>;

/**
* After
*/
type NativeProps = $ReadOnly<{|
    onDrawerStateChanged?: ?((state: DrawerStateEvent) => mixed), 
    children?: Node,
    drawerBackgroundColor: ColorValue,
    drawerLockMode?: ?("unlocked" | "locked-closed" | "locked-open"),
    drawerPosition: ?number,
    drawerWidth?: ?number,
    keyboardDismissMode?: ?("none" | "on-drag"),
    onDrawerClose?: ?(() => mixed),
    onDrawerOpen?: ?(() => mixed),
    onDrawerSlide?: ?((event: DrawerSlideEvent) => mixed),
    renderNavigationView: () => Element<any>,
    statusBarBackgroundColor?: ?ColorValue,
    style?: ?ViewStyleProp
  |}>
```


I'm also not sure about `onDrawerStateChanged`, should the type be `?((state: DrawerStateEvent) => mixed)`? 


I tried reading about how the `...$Diff` utility type works, but I'm still confused.

 If I got it wrong, any help would be welcome at this point 🙏

Reply:
Yep, exactly. Just like that!

Reply:
Feel free to leave it the way it is now, I actually have an internal change that resolves this issue as well and I'll land it as soon as it is reviewed :)